pkg/store/proxy.go

```diff
@@ -315,7 +311,14 @@ func startStreamSeriesSet(
 				s.warnCh.send(storepb.NewWarnSeriesResponse(errors.New(w)))
 				continue
 			}
-			s.recvCh <- r.GetSeries()
+
+			select {
+			case s.recvCh <- r.GetSeries():
+				continue
+			case <-ctx.Done():
+				return
```

Comment:
I think this should also take care of partial responses.

In the case of context cancel we want:
```
	case <-ctx.Done():

	wrapErr := errors.Wrapf(ctx.Err(), "receive series from %s", s.name)
	if partialResponse {
			s.warnCh.send(storepb.NewWarnSeriesResponse(wrapErr))					
			return				
	}			
	
	s.errMtx.Lock()					s.errMtx.Lock()
	s.err = wrapErr					s.err = wrapErr
	s.errMtx.Unlock()					s.errMtx.Unlock()
	return
```

Reply:
Hmm, we did not do it previously though... Let me look on the flow. We definitely want to show partial response OR fail whole thing in this case.

Reply:
yes! @GiedriusS Double checked and we need to:

Return error if partial response is disabled or send err to warn Ch (with who canelled as well somehow). (: I know your PR did not add the regression, but let's fix it?

Reply:
discussed offline all good.